### PR TITLE
[#164563148][#162826381] Make sign in/sign up page simple with Facebook login

### PR DESCRIPTION
**Pull request by tri-dang**
1. We need to update logo in our Facebook app, so happy if someone has access to our Facebook app do it for me or I can do that if I have access to our Facebook app
2. Not sure if we should remove all legacy codes related to registering new account by email, or reseting password, ...
3. Now admin need to login through Facebook too.

### CODE REVIEW

**Comment by tri-dang** [comment]
1. Stub sign_in in acceptance test
2. Groups tests into a file related logic
3. Add attachment to `Setting`. So I need to remove `Setting.site_title` in `active_admin` config. This lead to load `Setting` model before loading `active_storage` config, leads to `undefine method` error  of `has_on_attached`, we need to figure out how to fix this. But I will leave it now.
4. Use Setting.site_logo in sign_in/sign_up form

File: app/admin/items.rb [hunk 1]
```diff
@@ -49,7 +49,7 @@
 
       f.input :image, :hint => f.object.has_image? \
               ? f.image_tag(f.object.image_urls[:thumb], style: 'height: 150px;')
-              : f.content_tag(:span, '-')
+              : f.content_tag(:span, '-'), as: :file
```
Comment (tri-dang) [inline app/admin/items.rb]: this fix help to create upload image when creating new item in admin

File: app/models/setting.rb [hunk 2]
```diff
@@ -29,5 +34,21 @@ def self.site_tagline
     fetch 'site.tagline', 'GPS meets Photography meets Messaging'
   end
 
+  def self.site_logo
+    fetch 'site.logo', 'Logo of site', {
+      io: File.open(Rails.root.join('app', 'assets', 'images', 'logo.png')), filename: 'logo.png'
```
Comment (tri-dang) [inline app/models/setting.rb]: I add a default logo here. `value` is just a dummy text to pass validation.

File: app/models/setting.rb [hunk 3]
```diff
@@ -29,5 +34,21 @@ def self.site_tagline
...
+    }
+  end
+
+  def has_attachment?
+    attachment.attached?
+  end
+
+  def is_image_attached?
+    has_attachment? && attachment.content_type.include?('image')
```
Comment (tri-dang) [inline app/models/setting.rb]: `image/png` or `image/jpg`

**Comment by sillylogger** [comment]
Merged to master separately.

**Pull request closed without merging**